src/components/Evolution.ts

```diff
@@ -0,0 +1,92 @@
+import { Component } from "eightbittr/lib/Component";
+
+import { IPokemon } from "./Battles";
+
+import { FullScreenPokemon } from "../FullScreenPokemon";
+
+import { IPokemonEvolution, IPokemonEvolutionByHappiness, IPokemonEvolutionByItem, 
+    IPokemonEvolutionByLevel, IPokemonEvolutionByStats, IPokemonEvolutionByTime, 
+    IPokemonEvolutionByTrade, Pokemon } from "./constants/Pokemon";
+/**
+ * Holds things related to Pokemon evolution.
+ */
+export class Evolution<TGameStartr extends FullScreenPokemon> extends Component<TGameStartr> {
+    /**
+     * Acts according to what form of requirement is passed to it.
+     */
+    public requirementHandlers: IRequirementHandlers = {
+        "level": function (pokemon: IPokemon, requirements: IPokemonEvolutionByLevel): boolean {
+            return pokemon.level >= requirements.level;
+        },
+        "item": function (): boolean {
+            // Need some way to pass a flag for the item being used
+            // Item usage outside of battle does not seem to be implemented
```

Comment:
You should reference the actual issue number here, so it's easy to cross-reference.